hyperspy/models/edsmodel.py

```diff
@@ -0,0 +1,822 @@
+                    component.sigma.bmin = None
+                    component.sigma.bmax = None
+                elif component.name in xray_lines:
+                    component.sigma.twin = None
+                    component.sigma.free = False
+                    component.sigma.bmin = None
+                    component.sigma.bmax = None
+
+    def calibrate_xray_lines(self,
+                             calibrate='energy',
+                             xray_lines='all',
+                             bound=1,
+                             kind='single',
+                             **kwargs):
+        """
+        Calibrate individually the X-ray line parameters.
+
+        The X-ray line energy, the weight of the sub-lines and the X-ray line
+        width can be calibrated.
+
+        Parameters
+        ----------
+        calibrate: 'energy' or 'sub_weight' or 'width'
+            If 'energy', calibrate the X-ray line energy.
+            If 'sub_weight', calibrate the ratio between the main line
+            alpha and the other sub-lines of the family
+            If 'width', calibrate the X-ray line width.
+        xray_lines: list of str or 'all'
+            The Xray lines. If 'all', fit all lines
+        bounds: float
+            for 'energy' and 'width' the bound in energy, in eV
+            for 'sub_weight' Bound the height of the peak to fraction of
+            its height
+        kind : {'single', 'multi'}
+            If 'single' fit only the current location. If 'multi'
+            use multifit.
+        **kwargs : extra key word arguments
+            All extra key word arguments are passed to fit or
+            multifit, depending on the value of kind.
+        """
+
+        if calibrate == 'energy':
+            bound = bound / 1000. * self.units_factor
+            free = self.free_xray_lines_energy
+            fix = self.fix_xray_lines_energy
+        elif calibrate == 'sub_weight':
+            free = self.free_sub_xray_lines_weight
+            fix = self.fix_sub_xray_lines_weight
+        elif calibrate == 'width':
+            bound = bound / 1000. * self.units_factor
+            free = self.free_xray_lines_width
+            fix = self.fix_xray_lines_width
+
+        free(xray_lines=xray_lines, bound=bound)
+        if kind == 'single':
+            self.fit(bounded=True, fitter='mpfit', **kwargs)
+        elif kind == 'multi':
+            self.multifit(bounded=True, fitter='mpfit', **kwargs)
+        fix(xray_lines=xray_lines)
+
+    def get_lines_intensity(self,
+                            xray_lines=None,
+                            plot_result=False,
+                            **kwargs):
+        """
+        Return the fitted intensity of the X-ray lines.
+
+        Return the area under the gaussian corresping to the X-ray lines
+
+        Parameters
+        ----------
+        xray_lines: list of str or None or 'from_metadata'
+            If None, all main X-ray lines (alpha)
```

Comment:
Sure, I just thought it would be quick to add:

First, separate out the loop body in `get_lines_intensity` (after `for xray_line in xray_lines:`) as a new function (e.g. `_get_single_line_intensity(self, xray_line)`), then add the following method:
```python
def get_element_intensities(self, elements=None, **kwargs):
    if elements is None:
        elements = self.spectrum.metadata.Sample.elements
    xray_lines = self.spectrum._get_lines_from_elements(elements, only_lines=None)
    intensities = {}
    for xray_line in xray_lines:
        element, line = utils_eds._get_element_and_line(xray_line)
        i = self._get_single_line_intensity(xray_line)
        if element in intensities:
            intensities[element] += i
        else:
            intensities[element] = i
    return intensities
```

If you don't want it in, then I can split it off as a new PR.